opal/util/output.c

```diff
@@ -176,8 +176,7 @@ bool opal_output_init(void)
     } else {
         verbose.lds_want_stderr = true;
     }
-    gethostname(hostname, sizeof(hostname));
-    hostname[sizeof(hostname)-1] = '\0';
+    gethostname(hostname, MAXHOSTNAMELEN);
```

Comment:
Is this kind of stuff correct?

I ask because, for example, the OS X man page for `gethostname(2)` says:

>      Host names are limited in length to {sysconf(_SC_HOST_NAME_MAX)} characters, not including the trailing null, currently 255.

Since we're using `MAXHOSTNAMELEN` and not `sysconf()`, we can't know that it is big enough to include the terminating character.